### PR TITLE
Added ignore-sgx for appropriate tests in src/test

### PR DESCRIPTION
**Pull request by Goirad**
These are all the tests that make sense to ignore when targeting fortanix-unknonw-sgx, at least in test/runpass. Other suites not yet covered.

### CODE REVIEW

File: src/test/run-pass/cfg/cfg-family.rs [hunk 1]
```diff
@@ -10,3 +8,7 @@ pub fn main() {
 #[cfg(unix)]
 pub fn main() {
 }
+
+#[cfg(not(any(windows, unix)))]
```
Comment (rasendubi) [inline src/test/run-pass/cfg/cfg-family.rs]: Doesn't this miss the point of test? (Same for target_family test.)

Comment (jethrogb) [reply src/test/run-pass/cfg/cfg-family.rs]: What's your suggestion for making the test suite pass on non-windows non-unix platforms? It seems silly to me to add an `ignore` statement for every single such platform that may be added to Rust.

File: src/test/ui/rfc-2361-dbg-macro/dbg-macro-expected-behavior.rs [hunk 2]
```diff
@@ -69,44 +70,46 @@ fn test() {
...
         ")",
-        ":46] f(&42) = 42",
+        ":47] f(&42) = 42",
 
         "before",
-        ":51] { foo += 1; eprintln!(\"before\"); 7331 } = 7331",
+        ":52] { foo += 1; eprintln!(\"before\"); 7331 } = 7331",
 
-        ":59] (\"Yeah\",) = (",
+        ":52] { foo += 1; eprintln!(\"before\"); 7331 } = 7331",
```
[the remark this answers is not in the document]
Comment (Goirad) [reply src/test/ui/rfc-2361-dbg-macro/dbg-macro-expected-behavior.rs]: I botched the merge, it is fixed now. It is unfortunate that the line numbers have to be hardcoded into the test.

File: src/test/run-pass/atomic-print.rs [hunk 3]
```diff
@@ -30,9 +31,13 @@ fn main(){
         do_print(2);
         t.join();
     } else {
+        println!("{:?}", env::args());
```
Comment (nikomatsakis) [inline src/test/run-pass/atomic-print.rs]: maybe remove these stray printlns?

File: src/test/run-pass/cfg/cfg-family.rs (hunk 1 above)
Comment (nikomatsakis) [reply src/test/run-pass/cfg/cfg-family.rs]: what exactly *is* the point of this test? I guess it's to test that you have *either* windows *or* unix (except when you don't, hence the ignores?)

In that case, perhaps we want to add some kind of "meta ignore" statement instead?

Comment (nikomatsakis) [reply src/test/run-pass/cfg/cfg-family.rs]: But, honestly, adding some ignore statements doesn't seem terrible for the time being. =) How often do we add such platforms anyway? And it's only two more tests.

File: src/test/run-pass/atomic-print.rs (hunk 3 above)
Comment (nikomatsakis) [reply src/test/run-pass/atomic-print.rs]: oh, I see that is done in a later commit

File: src/bootstrap/bootstrap.py [hunk 4]
```diff
@@ -678,7 +678,7 @@ def update_submodule(self, module, checked_out, recorded_submodules):
         run(["git", "submodule", "-q", "sync", module],
             cwd=self.rust_root, verbose=self.verbose)
         run(["git", "submodule", "update",
-            "--init", "--recursive", "--progress", module],
+            "--init", "--recursive", module],
```
Comment (nikomatsakis) [inline src/bootstrap/bootstrap.py]: why this change?

File: src/bootstrap/bootstrap.py [hunk 5]
```diff
@@ -678,7 +678,7 @@ def update_submodule(self, module, checked_out, recorded_submodules):
         run(["git", "submodule", "-q", "sync", module],
             cwd=self.rust_root, verbose=self.verbose)
         run(["git", "submodule", "update",
-            "--init", "--recursive", module],
+            "--init", "--recursive", "--progress", module],
```
Comment (nikomatsakis) [inline src/bootstrap/bootstrap.py]: Nit: it'd be nice to use git rebase -i to remove this from the history altogether (same with the printlns)

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
This seems good but:

- the git history needs a bit of cleanup before we can merge this. It'd be nice to remove (e.g.) the stray printlns from the history and other such accidents. We also avoid merging PRs with merge commits. Are you familiar with `git rebase`?
- I agree with @rasendubi that the `#[cfg(not(any(windows, unix))]` annotations seem to defeat the point of the test. I'm not sure what's a better more general fix, but I think for now it makes sense to just add a `// ignore-foo` annotations (after all, we do it on a ton of other tests, so I don't see why it's important not to do it on this particular test, particularly if we don't have a compelling alternative).

File: src/test/run-pass/cfg/cfg-family.rs (hunk 1 above)
Comment (jethrogb) [reply src/test/run-pass/cfg/cfg-family.rs]: > what exactly *is* the point of this test?

I think it's to test that `cfg(unix)` works, as in, that the compiler correctly sets the configuration based on the target specification.

File: src/bootstrap/bootstrap.py (hunk 5 above)
Comment (Goirad) [reply src/bootstrap/bootstrap.py]: I would absolutely like to tidy up the git history, but I am not familiar enough with git to really know how to start. I am know about rebase, but I need to do some more learning to be able to use it to clean up a git history. I'll get right on it.

Comment (jethrogb) [reply src/bootstrap/bootstrap.py]: @Goirad I'll talk to you offline

File: src/test/run-pass/atomic-print.rs (hunk 3 above)
Comment (Goirad) [reply src/test/run-pass/atomic-print.rs]: I have cleaned up the git history

File: src/test/run-pass/cfg/cfg-family.rs (hunk 1 above)
Comment (nikomatsakis) [inline src/test/run-pass/cfg/cfg-family.rs]: I still this [@rasendubi was correct about this change defeating the point of the test](https://github.com/rust-lang/rust/pull/60386/commits/48f5b70e9055540622972cee05596ea60e0e9176#r279743858); I don't see how it could fail under this setup. Therefore, I'm not inclined to merge until that is addressed in some way. But the rest seems great!

File: src/test/run-pass/cfg/cfg-target-family.rs [hunk 6]
```diff
@@ -11,3 +9,7 @@ pub fn main() {
 #[cfg(target_family = "unix")]
 pub fn main() {
 }
+
+#[cfg(not(any(target_family = "windows", target_family = "unix")))]
```
Comment (nikomatsakis) [inline src/test/run-pass/cfg/cfg-target-family.rs]: Similarly here

**Review by nikomatsakis — CHANGES_REQUESTED** [review]
As far as I can tell, those two tests still ought to be changed.

cc @alexcrichton -- I suspect you know something about those tests, do you agree these changes are defeating the point of the tests? I'm not 100% sure.

**Comment by alexcrichton** [comment]
It looks like the original changes aren't here any more (the tests have been updated). FWIW our test suite really wasn't ever written (or will be...) to run on every single platform under the sun. As can be seen from the other annotations in these files getting the test suite running on a particular platform involves just sprinkling a lot of directives throughout the codebase, and naturally this will regress quite often since we're not testing it on CI itself. I wish we had a better system for managing this but we aren't currently employing anything.

**Comment by Goirad** [comment]
@nikomatsakis I have reverted those tests to just adding the ignore directives.

**Review by nikomatsakis — APPROVED** [review]
r=me

**Comment by nikomatsakis** [comment]
@Goirad ok thanks! Sorry for the delay. r=me once rebased.

**Comment by Goirad** [comment]
@nikomatsakis That's alright! And everything is rebased
